diff --git a/node_modules/next-auth/dist/client/index.js b/node_modules/next-auth/dist/client/index.js
index 773ec1e..255b99d 100644
--- a/node_modules/next-auth/dist/client/index.js
+++ b/node_modules/next-auth/dist/client/index.js
@@ -296,7 +296,9 @@ var signIn = function () {
       };
       var res = yield fetch(signInUrl, fetchOptions);
       var data = yield res.json();
-      window.location = data.url ? data.url : callbackUrl;
+      if(args.disableCallback !== true){
+        window.location = data.url ? data.url : callbackUrl;
+      }
     }
   });
 
@@ -333,7 +335,9 @@ var signOut = function () {
       }
     });
 
-    window.location = data.url ? data.url : callbackUrl;
+    if(args.disableCallback !== true){
+      window.location = data.url ? data.url : callbackUrl;
+    }
   });
 
   return function signOut() {
